pkg/ipcache/metadata.go

```diff
@@ -221,36 +246,58 @@ func (ipc *IPCache) InjectLabels(ctx context.Context, modifiedPrefixes []netip.P
 			// kvstore and via the k8s control plane. If the new
 			// security identity is the same as the one currently
 			// being used, then no need to update it.
-			if id.ID == newID.ID {
+			if oldID.ID == newID.ID {
 				goto releaseIdentity
 			}
 
-			idsToAdd[newID.ID] = lbls.LabelArray()
-			entriesToReplace[prefix] = Identity{
-				ID:     newID.ID,
-				Source: prefixInfo.Source(),
-			}
+			// Overridden identities do not get added to the policy map and
+			// should not be released when the IPCache entry is removed
+			override := overrideID.IsValid()
+
 			// IPCache.Upsert() and friends currently require a
 			// Source to be provided during upsert. If the old
 			// Source was higher precedence due to labels that
 			// have now been removed, then we need to explicitly
 			// work around that to remove the old higher-priority
 			// identity and replace it with this new identity.
-			if entryExists && prefixInfo.Source() != id.Source {
-				forceIPCacheUpdate[prefix] = true
+			force := entryExists && prefixInfo.Source() != oldID.Source
+
+			entriesToReplace[prefix] = entry{
+				Identity: Identity{
+					ID:     newID.ID,
+					Source: prefixInfo.Source(),
+				},
+				override: override,
+				force:    force,
+			}
+
+			if !override {
+				// We allocated a new identity, which means we want to add it
+				// to the policy map
```

Comment:
Absolutely agree. At the moment, there is no separate method to inject override identities, but it might make sense to have one even just for documentation purposes.